### PR TITLE
Removed many global imports from std.experimental.logger.core

### PR DESCRIPTION
**Pull request by JackStouffer**

### CODE REVIEW

File: std/experimental/logger/core.d [hunk 1]
```diff
@@ -1,17 +1,9 @@
...
 module std.experimental.logger.core;
 
-import std.array;
-import std.stdio;
-import std.conv;
 import std.datetime;
-import std.string;
-import std.range;
+import std.range.primitives;
 import std.traits;
```
Comment (ZombineDev) [inline std/experimental/logger/core.d]: Use selective imports?

Comment (JackStouffer) [inline std/experimental/logger/core.d]: There's really no benefit to top-level selective imports at the moment. DMD still scans the entire imported file; all selective imports do is change visibility rules. This is something that can and should change (SDC does the right thing), and when it does, I will gladly fix this.

Plus, selective top level imports aren't the current style in the rest of Phobos.

Comment (ZombineDev) [inline std/experimental/logger/core.d]: The biggest benefit for me personally is that they document explicitly what is being imported. This makes them easier to convert to convert to local imports and to whatever DIP1005 brings to the table (language change or something along the lines of https://github.com/dlang/druntime/pull/1756).

> Plus, selective top level imports isn't the current style in the rest of Phobos.

Where we are at the moment is not interesting to talk about. What matters is where we want to be and how to get there.

Comment (joakim-noah) [inline std/experimental/logger/core.d]: There is a benefit right now: the compiler has to search all the top-level modules and any publicly imported modules they reference to see where a symbol is invoked from and check for name collisions.  When you specify the module a symbol comes from, that search is skipped.

The real issue is that we have held off on using module-scope selective imports until we're sure the symbol leak bugs are fixed and downstream code has adapted.  Now that 2.073 has shipped, maybe @MartinNowak will OK their use?

**Comment by JackStouffer** [comment]
This debate isn't dependent on the changes here and can be done in a separate PR; this can be merged as is.

**Comment by ZombineDev** [comment]
> This debate isn't dependent on the changes here and can be done in a separate PR; this can be merged as is.

Hence, why I first approved your PR ;)

**Comment by joakim-noah** [comment]
>Yep, but that issue is independent from this PR. How about opening a NG discussion?

It is not completely independent, the only reason he's not using selective imports in this PR is because of this long-standing convention.  I agree that it doesn't matter if this PR uses them or not, so I'm not suggesting that this PR wait on that.

**Comment by JackStouffer** [comment]
Auto-merge toggled on

**Pull request merged**